### PR TITLE
Fix leap year bug

### PR DESCRIPTION
**Pull request by theonefx**
# Description

## Issue reference
https://github.com/dapr/dapr/issues/3995

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -1648,7 +1648,11 @@ func parseISO8601Duration(from string) (time.Duration, int, error) {
 		}
 		switch name {
 		case "year":
-			duration += time.Hour * 24 * 365 * time.Duration(val)
+			daysPreYear := time.Duration(365)
+			if ((val%4) == 0 && (val%100) != 0) || (val%400) == 0 {
+				daysPreYear = time.Duration(366)
```
Comment (daixiang0) [inline pkg/actors/actors.go:1653]: Please use `time.YearDay()`.

Comment (theonefx) [reply pkg/actors/actors.go:1653]: The "val" is not a specific year, but a number of years. So, it's not the correct solution....

File: tests/e2e/actor_sdks/actor_sdks_test.go [hunk 2]
```diff
@@ -172,8 +172,8 @@ func TestActorInvocationCrossSDKs(t *testing.T) {
 		healthCheckApp(t, externalURL, numHealthChecks)
 	}
 
-	t.Log("Sleeping for 10 seconds ...")
-	time.Sleep(10 * time.Second)
+	t.Log("Sleeping for 15 seconds ...")
```
Comment (daixiang0) [inline tests/e2e/actor_sdks/actor_sdks_test.go:175]: I am afraid that sleep is not a good solution :(

Comment (theonefx) [reply tests/e2e/actor_sdks/actor_sdks_test.go:175]: Just my guess, to solve part of the e2etest doesn't pass. But it seems not working. I will roll back this change.

**Comment by theonefx** [comment]
Close this PR and resubmit a new one after resolving the DCO

**Comment by theonefx** [comment]
the new PR: https://github.com/dapr/dapr/pull/4027

**Pull request closed without merging**